ATOMIC_SWAP.md

```diff
@@ -0,0 +1,335 @@
+# Atomic Swap
+
+This is a how-to of using the atomic-swap feature of the wallet.
+Both P2PKH and P2SH (MultiSig) can participate on the atomic-swap.
+
+## Description
+
+An atomic-swap transaction is a transaction with multiple tokens and inputs from different wallets.
+The atomic-swap process is how participants can safely coordinate the content and signing of a transaction.
+
+## Atomic-swap process
+
+The process can be divided in 2 phases:
+1. Negotiation phase
+  - Participants will coordinate the content of the atomic-swap.
+1. Signing phase
+  - Participants will sign the proposed transaction and exchange the signatures.
+  - A participant will collect the signatures and push the signed transaction on the network.
+
+### Simulation
+
+Let's divide the steps from an atomic-swap using a simulated exchange between Alice and Bob.
```

Comment:
I like the example but I feel we should have the requests detailed in all the steps. After this section you start describing all the requests but they could be detailed in each example step.

Reply:
I like the granularity level the current simulation section has. It is long enough to have relevant information but not deep enough to scare developers that are having their first contacts with the feature ( possibly even with the concept of an atomic swap ).

Still, I agree that a list with a more low-level explaining each request could be helpful. For this purpose, what do you think of having an integrated test code embedded on the MD file?
The GitHub syntax for this is just the file URL with the lines specified:
```
https://github.com/HathorNetwork/hathor-wallet-headless/blob/df255e9f0f1126d7488962381a1e5a723026c452/__tests__/integration/atomic-swap.test.js#L56-L125
```

And the result is rendered like the one below:
https://github.com/HathorNetwork/hathor-wallet-headless/blob/df255e9f0f1126d7488962381a1e5a723026c452/__tests__/integration/atomic-swap.test.js#L56-L125

Deeper than that, maybe a sample application on our GitHub with a very commented code connecting to the headless could be a good idea.